plugins/com.gwtplugins.gwt.eclipse.core/src/com/google/gwt/eclipse/core/modules/ModuleFile.java

```diff
@@ -1,374 +1,420 @@
+        model.changedModel();
+
+        // Save the model if necessary
+        if (!model.isSharedForEdit() && model.isSaveNeeded()) {
+          model.save();
+        }
+      } catch (IOException e) {
+        GWTPluginLog.logError(e);
+      } catch (CoreException e) {
+        GWTPluginLog.logError(e);
+      } finally {
+        if (model != null) {
+          model.releaseFromEdit();
+        }
+      }
+    }
+
+    protected abstract void editModel(IDOMModel model);
+
+    private IDOMModel getModelForEdit() throws IOException, CoreException {
+      IModelManager modelManager = StructuredModelManager.getModelManager();
+      return (IDOMModel) modelManager.getModelForEdit(getFile());
+    }
+  }
+
+  private String qualifiedName = null;
+  private final IFile storage;
+
+  protected ModuleFile(IFile file) {
+    this.storage = file;
+  }
+
+  /**
+   * Adds a new entry point to a module.
+   *
+   * @param qualifiedTypeName name of the entry point class, which should
+   *        com.google.gwt.core.client.EntryPoint
+   * @throws Exception if there was a problem modifying or saving the module XML
+   */
+  public void addEntryPoint(final String qualifiedTypeName) throws Exception {
+    new EditModelOperation() {
+      @Override
+      public void editModel(IDOMModel model) {
+        IDOMDocument editDoc = model.getDocument();
+
+        // TODO: create an empty element (no closing </entry-point>)
+
+        Element entryPointElement = editDoc.createElement(ENTRY_POINT_TAG_NAME);
+        entryPointElement.setAttribute(CLASS_ATTRIBUTE_NAME, qualifiedTypeName);
+        editDoc.getDocumentElement().appendChild(entryPointElement);
+
+        // Reformat the XML source to keep it nice and neat
+        new FormatProcessorXML().formatModel(model);
+      }
+    }.run();
+  }
+
+  /**
+   * Returns the package name for the module.
+   */
+  @Override
+  public String getQualifiedName() {
+    // Cache the qualified name
+    if (qualifiedName == null) {
+      String shortName = getShortName();
+      if (shortName != null && !shortName.isEmpty()) {
+        qualifiedName = getGwtMaven2ModuleName();
+        return qualifiedName;
+      }
+
+      qualifiedName = Util.removeFileExtension(storage.getName());
+
+      String modulePckg = doGetPackageName();
+      if (modulePckg != null) {
+        qualifiedName = modulePckg + "." + qualifiedName;
+      }
+    }
+
+    return qualifiedName;
+  }
+
+  /**
+   * GWT Maven project will have a short name
+   * @return the short name for module
+   */
+  private String getShortName() {
+    IFile file = storage;
+    IFolder moduleFolder = (IFolder) file.getParent();
+    String shortName = WebAppProjectProperties.getGwtMavenModuleShortName(moduleFolder.getProject());
+    return shortName;
+  }
+
+  /**
+   * Get the gwt maven2 module name
+   * @return module name
+   */
+  private String getGwtMaven2ModuleName() {
+    IFile file = storage;

```

Comment:
This is silly to have, and the moduleName variable isnt needed either. Probably should just inline it (but the code is so much better I'd be okay with leaving it).